Code/Framework/AzCore/Tests/Serialization/Json/AnySerializerTests.cpp

```diff
@@ -0,0 +1,281 @@
+            AZStd::any anyT;
+            anyT = instanceT;
+            return AZStd::make_shared<AZStd::any>(anyT);
+        }
+
+        AZStd::shared_ptr<AZStd::any> CreatePartialDefaultInstance()
+        {
+            T instanceT = *T::GetInstanceWithSomeDefaults().m_instance.get();
+            AZStd::any anyT;
+            anyT = instanceT;
+            return AZStd::make_shared<AZStd::any>(anyT);
+        }
+
+        std::optional<AZStd::string_view> GetJSON([[maybe_unused]] const JSONRequestSpec& requestSpec) override
+        {
+            if (requestSpec.jsonStorage == DefaultJSONStorage::Store)
+            {
+                if (requestSpec.objectSupplied == DefaultObjectSuppliedInSerialization::True)
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        // fully configured
+                    }
+                }
+                else
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        // fully configured
+                    }
+                }
+            }
+            else
+            {
+                // dropping defaults
+                if (requestSpec.objectSupplied == DefaultObjectSuppliedInSerialization::True)
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+                        
+                    }
+                    else
+                    {
+                        // fully configured
+                        return m_jsonForFullyConfiguredInstanceWithoutDefaults.begin();
+                    }
+                }
+                else
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        return m_jsonForFullyConfiguredInstanceWithoutDefaults.begin();
+                    }
+                }
+            }
+
+            return AZStd::nullopt;
+        }
```

Comment:
First the return type should be AZStd::optional.
Second, most of this function is empty if blocks and should be reduced.